invite/test/invitation_record.test.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2020 The AMP HTML Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Invite, InviteAction} from '../src/types';
+import {InvitationRecord} from '../src/invitation_record';
+
+describe.skip('invitation record', () => {
+  let record: InvitationRecord;
+  const invite: Invite = {
+    username: 'someone',
+    repo: 'test_repo',
+    issue_number: 1337,
+    action: InviteAction.INVITE,
+  };
+  const archivedInvite: Invite = Object.assign({archived: true}, invite);
+
+  beforeEach(() => {
+    record = new InvitationRecord();
+  });
+
+  describe('recordInvite', () => {
+    // TODO: Update once the storage interface is determined.
+    it('records an invite', async done => {
```

Comment:
`it.todo('records an invite');`

Reply:
> Is storage really necessary? You can probably query the API to see whether a user is pending an invitation, right?

This is to record invites triggered by comments. For example, suppose Joey comments `/tryassign @somebody` on two issues and `/invite @somebody` on a third. When that invite is eventually accepted, we want to assign those two issues to `@somebody` and add comments on all three updating the thread that the invited user has joined. This metadata isn't available in the invite, and so the mapping to issues from which users were invited needs to be persisted.

Reply:
Counterpoint - can you use the API to search for all comments containing `/invite @somebody` and `/tryassign @somebody` and work your way from there? 🙃
Stateless code is code that's easier to debug and reason about, so I always prefer to try that

Reply:
a) Search is limited to one repository, while this invite-bot will operate organization-wide
b) Search API is ugly and nondeterministic and has no consistent ordering, which completely breaks pagination (to find all OWNERS files with the search API, I literally just have to try multiple times and hope I got them all)
c) Persisting the invites to act on also acts as a record log for debugging and reasoning about the actions the bot takes

Reply:
Fair. Proceed!

Reply:
I shall! Iff you approve :upside_down_face: